### PR TITLE
DM-47655: Add 'Time since last image' clock

### PR DESCRIPTION
**Pull request by ugyballoons**
Add a clock for ComCam and ComCamAOS that shows the time since the last image on both the table and current event pages (same for all channels).

### CODE REVIEW

**Review by sebastian-aranda — CHANGES_REQUESTED** [review]
Good job @ugyballoons, this was a big PR so I left several comments around. Please take a look when you can :pray:.

File: src/js/modules/utils.js [hunk 1]
```diff
@@ -80,29 +80,29 @@ export function indicatorForAttr (attributes, attrToCheck) {
...
 
-export function sanitiseString (str) {
+export function sanitiseString(str) {
   // Substitutes spaces and hyphens are exchanged for underscores.
   // Any non-word characters (any but a-z, A-Z, _) are removed.
   // Capital letters are made small.
-  let sanitised = str.replace('/[\s-]/', '_')
-  sanitised = sanitised.replace('[\W]','')
+  let sanitised = str.replace("/[s-]/", "_")
+  sanitised = sanitised.replace("[W]", "")
```
[the remark this answers is not in the document]
Comment (ugyballoons) [reply src/js/modules/utils.js:88]: Thanks for spotting that! I installed the pretty-eslint formatter and then clearly didn't properly check where it had replaced double for single quotes. I've removed the quotes entirely and replaced `str.replace()` with `str.replaceAll()`

File: src/js/components/Clock.js [hunk 2]
```diff
@@ -16,21 +16,118 @@ export default function Clock () {
...
+      .pop()
+    row = metadata[lastSeq]
+  }
+  if (!row) {
+    return
+  }
+  const toSum = ["Date begin", "Exposure time"]
+  let error, timeElapsed
+  if (
+    !toSum.reduce((acc, col) => Object.keys(row).includes(col) && acc, true)
```
[the remark this answers is not in the document]
Comment (ugyballoons) [reply src/js/components/Clock.js:84]: That's quite a lot clearer!

File: src/js/hostbanner.js [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
 
-      const heartbeatWsUrl = getWebSockURL("ws/heartbeats")
-      heartbeatWorker.port.postMessage({heartbeatWsUrl})
+    const heartbeatWsUrl = getWebSockURL("ws/heartbeats")
+    heartbeatWorker.port.postMessage({ heartbeatWsUrl })
 
-      heartbeatWorker.port.onmessage = function(e) {
-        console.log("message from worker:", e.data)
-      }
+    heartbeatWorker.port.onmessage = function (e) {
```
[the remark this answers is not in the document]
Comment (ugyballoons) [reply src/js/hostbanner.js:29]: Your're right. Instead of fighting with prettier, this is now an arrow function.

**Review by sebastian-aranda — APPROVED** [review]
Thanks for the changes @ugyballoons ! PR approved :+1:.

**Pull request merged**